pkg/traceql/engine.go

```diff
@@ -47,38 +75,19 @@ func (e *Engine) Execute(ctx context.Context, searchReq *tempopb.SearchRequest,
 		// TODO capture and update metrics
 		Metrics: &tempopb.SearchMetrics{},
 	}
-
-	spansetsEvaluated := 0
-
-iter:
 	for {
-		spanSet, err := iterator.Next(ctx)
-		if err != nil {
-			span.LogKV("msg", "iterator.Next", "err", err)
-			return nil, err
-		}
-
-		if spanSet == nil {
+		spanset, err := iterator.Next(ctx)
+		if spanset == nil {
```

Comment:
nit:  Should it check err first?  I think this could lose the error if the return is `nil, err`  Might need to ignore `io.EOF`

Reply:
agreed. checking err first. i'm unsure on io.EOF, but it won't hurt to add it.